### PR TITLE
Update

### PR DESCRIPTION
**Pull request by itemniner**
## Type of change made:
- [ ] **Bugfix** (non-breaking change which fixes an issue)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [x] **Refactoring** (improves existing functionality)
- [ ] **Testing** (tests existing code or improves existing tests)
- [ ] **Styling** (no functionality changed, just visual improvements)
## Description:
- Creates Propublica Service for refactoring routes/controller.
- Refactored App file to use Propublica services to get API response and parse data for Users Reps endpoint.
- Refactored Members by ID endpoint to use Propublica Service to get API response and parse data.
- Refactored Members by State endpoint to use Propublica Service to get API response and parse data.
- Update Members show method to parse Propublica API response.
- Restructured the initialize method for the Members Show class.
## What functionality does this accomplish?
- Refactoring 
closes # | does not close anything
## Checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have fully tested my code
- [x] I have partially tested my code (please explain why): Now that we are using fixtures we need to update the test to be more specific in the assertions.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 db = SQLAlchemy(app)
 
 from models import User
+from services.propublica import PropublicaService
 from classes import MemberIndex, MemberShow
```
Comment (SMJ289) [inline app.py:20]: Nice! It is good to have this refactored to a service class!

File: app.py [hunk 2]
```diff
@@ -68,13 +69,8 @@ def get_by_id(id_):
...
-    URL_SENATE = f'https://api.propublica.org/congress/v1/members/senate/{state_}/current.json'
-    URL_HOUSE = f'https://api.propublica.org/congress/v1/members/house/{state_}/current.json'
-    senate_response = requests.get(URL_SENATE, headers = headers).json()
-    house_response = requests.get(URL_HOUSE, headers = headers).json()
-    senate_results = senate_response['results']
-    house_results = house_response['results']
+    senate_results = PropublicaService.senators_by_state(state_)
+    house_results = PropublicaService.reps_by_state(state_)
 
     senator_objects = map(lambda result: MemberIndex(id = result['id'],
```
Comment (SMJ289) [inline app.py:75]: Nice this looks much cleaner!

File: app.py [hunk 3]
```diff
@@ -89,13 +85,13 @@ def get_members_by_state(state_):
                                                  role = result['role'],
                                                  last_name =result['last_name']),
                                                  house_results)
-
```
Comment (SMJ289) [inline app.py:92]: Now we need to refactor the MemberIndex initialize method in the same way that we did the MemberShow initialize.

File: classes.py [hunk 4]
```diff
@@ -1,39 +1,25 @@
...
-        self.state = state
-        self.district = district
-        self.website = website
-        self.contact_form_url = contact_form_url
-    
-    def party_name(self, party):
+    def __init__(self, data):
+        self.id = data['id']
+        self.image = f"https://theunitedstates.io/images/congress/original/{data['id']}.jpg"
+        self.first_name = data['first_name']
```
Comment (SMJ289) [inline classes.py:5]: We might be able to refactor this little image URL into a small method.

**Comment by jklich151** [comment]
Refactoring of the api calls. This is much more abstracted.

File: services/propublica.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    def reps_by_state(state_):
+        headers = {'X-API-Key': os.getenv('PROP_API')}
+        URL = f'https://api.propublica.org/congress/v1/members/house/{state_}/current.json'
+        return (requests.get(URL, headers=headers).json())['results']
+
+    def reps_by_district(state_, district_):
+        headers = {'X-API-Key': os.getenv('PROP_API')}
+        URL = f'https://api.propublica.org/congress/v1/members/house/{state_}/{district_}/current.json'
+        return (requests.get(URL, headers=headers).json())['results']
+
```
Comment (SMJ289) [inline services/propublica.py:33]: This looks good but I think we should look into how to make these into class methods.

**Pull request merged**